sdk/storage/Azure.Storage.Files.DataLake/src/Models/AccessControlChangeOptions.cs

```diff
@@ -26,5 +28,20 @@ public class AccessControlChangeOptions
         /// user failures and proceed with the operation on other sub-entities of the directory.
         /// </summary>
         public bool? ContinueOnFailure { get; set; }
+
+        /// <summary>
+        /// Optional. <see cref="Progress{AccessControlChanges}"/> callback where caller can track progress of the operation
+        /// as well as collect paths that failed to change Access Control.
+        /// </summary>
+        public IProgress<Response<AccessControlChanges>> ProgressHandler { get; set; }
+
+        /// <summary>
+        /// Optional. If set to true and the operation contains failures, the operation will populate the result,
+        /// <see cref="AccessControlChangeResult.BatchFailures"/>, with a batch of failures.
+        /// If <see cref="ContinueOnFailure"/> is set to true, the first batch of failures will be populated in
+        /// result, otherwise the last batch of failures will be set.
+        /// This will default to false.
+        /// </summary>
+        public bool? RetrieveBatchFailures { get; set; }
```

Comment:
I'd remove that switch and just do what it describes implicitly. If we ask user to set this flag we can equally ask to set progress handler. The idea is to get first/last batch of failures out of the box on default settings.
Btw. in case continueOnFailure = false last batch is same as first batch, so we could say that it's first batch where failures where encountered.